pep-0517.txt

```diff
@@ -218,6 +241,14 @@ metadata. The directory passed in by the build frontend MUST be
 identical to the directory created by ``get_wheel_metadata``,
 including any unrecognized files it created.
 
+This method only places the *contents* of the wheel file, without actually
+creating the final artifact. This is to allow flexibility in how the result is
+going to be used. A front end like pip, which is only going to immediately
+install this wheel can speed up the installation process by skipping a round
+trip through zip. However a command like a hypothetical ``twine wheel`` could
```

Comment:
> From experience with conda (which uses hard linking much like you suggest), I'd be wary of this. I've seen files get modified in one installed version, but because it's hard linked, that modification affects every installed copy - and it's in the cache, so even reinstalling doesn't get the original version. When we suspect this, we usually advise people to uninstall and reinstall the whole Anaconda distribution.

I guess "_switching to_" is a strong expression. I was more thinking in the line of:

 - having a switch to pip to cache wheels unpacked, and 
 - having another switch to use symlinks or hardlinks when installing (hardlinks can be create unpriviledged on Windows, while symlinks cannot). When missing, handle unpacked wheels by copying.

Like any pip switches, these switches could be persisted in a `pip.conf` file site-wide, user-wide or virtualenv-wide.